cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -288,18 +281,11 @@ else()
   # See cpp/build_support/trim-boost.sh
   set(
     BOOST_SOURCE_URL
+    "https://github.com/boostorg/boost/archive/boost-${ARROW_BOOST_BUILD_VERSION}.tar.gz"
```

Comment:
This should not be first. The one below this is what I added in ARROW-8222, 90% smaller file size only containing the bits we need.

Reply:
Understood, will demote

Reply:
@nealrichardson PTAL https://github.com/apache/arrow/pull/6825/files#diff-2420b0c5b6bdad921f1d538f92d64b59L288-R276

Reply:
How does https://github.com/ursa-labs/thirdparty/releases/download/latest/ get populated? IMO "https://dl.bintray.com/ursalabs/arrow-boost/boost_${ARROW_BOOST_BUILD_VERSION_UNDERSCORES}.tar.gz" should still be first because the [build scripts](https://github.com/apache/arrow/blob/master/cpp/build-support/upload-boost.sh) instruct you on uploading to bintray, so it would be surprising if you did that and then didn't see the change reflected in the boost_ep because the place we updated is actually #2 on the list to check.

Reply:
I'll address this in https://issues.apache.org/jira/browse/ARROW-6407

Ideally all of the ursalabs-managed archives should be pushed to by one script, not just boost. For now I'll add a comment in upload-boost.sh

Reply:
FWIW, it might be nice to move these auxiliary archives to a non-organization-specific GitHub repo. I created https://github.com/apache-arrow-dev and we could add all committers as maintainers there. @kszucs we could also move ursalabs/crossbow there